pkg/reconciler/ingress/ingress.go

```diff
@@ -67,8 +67,16 @@ func (reconciler *Reconciler) Reconcile(ctx context.Context, key string) error {
 	ingress := original.DeepCopy()
 	ingress.SetDefaults(ctx)
 	ingress.Status.InitializeConditions()
-
-	if err := reconciler.updateIngress(ingress); err != nil {
+	// This is the generation we are going to handle during this reconciliation loop.
+	ingress.Status.ObservedGeneration = ingress.GetGeneration()
+
+	err = reconciler.updateIngress(ingress)
+	if err == generator.ErrDomainConflict {
+		// If we had an error due to a duplicated domain, we must mark the ingress as failed with a
+		// custom status. We don't want to return an error in this case as we want to update its status.
+		reconciler.logger.Errorw(err.Error(), ingress.Name, ingress.Namespace)
+		ingress.Status.MarkLoadBalancerFailed("DomainConflict", "Ingress rejected as its domain conflicts with another ingress")
```

Comment:
Will this be retried at some point or the ingress stays there forever marked as invalid? :thinking:

Reply:
A global resync would retry it, I guess?

Reply:
Yes, the global resync will retry it. Do we want that to happen?

Reply:
Yep, I think that's fine. If the conflict is resolved this may pass happily.